test/e2e/rule_test.go

```diff
@@ -158,19 +160,26 @@ func reloadRulesSignal(t *testing.T, r e2e.InstrumentedRunnable) {
 }
 
 func checkReloadSuccessful(t *testing.T, ctx context.Context, endpoint string, expectedRulegroupCount int) {
-	req, err := http.NewRequestWithContext(ctx, "GET", "http://"+endpoint+"/api/v1/rules", ioutil.NopCloser(bytes.NewReader(nil)))
-	testutil.Ok(t, err)
-	resp, err := http.DefaultClient.Do(req)
-	testutil.Ok(t, err)
-	testutil.Equals(t, 200, resp.StatusCode)
+	var data = rulesResp{}
+	testutil.Ok(t, runutil.Retry(5*time.Second, ctx.Done(), func() error {
+		req, err := http.NewRequestWithContext(ctx, "GET", "http://"+endpoint+"/api/v1/rules", ioutil.NopCloser(bytes.NewReader(nil)))
+		testutil.Ok(t, err)
```

Comment:
Do we need to return the `err` here? Because `testutil.Ok(t, err)` will fail the whole test if `err` is not `nil`, I think :thinking:

Reply:
I copy pasted the contents into `runutil.Retry`. Let me fix! 🙂

Reply:
Checked all errors in https://github.com/thanos-io/thanos/pull/5558/commits/58138ffd0b709477f5801dbb5cc7950d444c2c46